src/Scrabble/Move/Move.hs

```diff
@@ -41,12 +41,40 @@ createMove' :: Validator -- ^
            ->  Either String Move
 createMove' validate b (Rack rack) wp dict = if valid then go else errMsg where
   errMsg        = Left "error: rack missing input letters"
-  rackLetters   = fmap letter rack
-  valid         = containsAll (toString putLetters) (toString rackLetters)
-  putLetters    = letter <$> wordPutTiles wp
-  rackRemainder = fmap fromLetter $ foldl' (flip delete) rackLetters putLetters
+  rackLetters   = toString (fmap letter rack)
+  valid         = containsAllWithBlanks putLetters rackLetters
+  putLetters    = toString (letter <$> wordPutTiles wp)
   go = do (newBoard, score) <- wordPut validate b wp dict
-          return $ Move wp score (Rack rackRemainder) newBoard
+          return $ Move wp score (rackRemainder (Rack rack) wp) newBoard
+
+
+-- | Like containsAll from Scrabble.Search, but if it encounters a character
+--   in s1 not in s2, it deletes a blank and tries again before returning false
+containsAllWithBlanks :: String -> Search
+containsAllWithBlanks s1 s2 = fst $ foldl' f (True, ups s2) (ups s1) where
+  f (b,s) c = if elem c s then (b, delete c s)
+              else if elem '_' s then (b, delete '_' s)
+              else (False, s)
+
+
+-- | Similar to containsAllwithBlanks, but it leaves the rack and wordPut contex
+--   to simplify the iteration.
+rackRemainder :: Rack -> WordPut -> Rack
+rackRemainder (Rack r) (WordPut tps) = Rack $ foldl' f r tps
+    where
+      rackLetters = map letter r
+
+      -- Get the tile associated with a TilePut so you can delete it in f
+      toTile :: TilePut -> Tile
+      toTile (LetterTilePut t _) = t
+      toTile (BlankTilePut _ _)  = fromLetter Blank
```

Comment:
if this doesn't already exist somewhere id be surprised. but if it doesn't, then can we move it to where it would best make sense?